src/rules/no-restricted-paths.js

```diff
@@ -53,9 +61,32 @@ module.exports = {
         }
 
         matchingZones.forEach((zone) => {
+          const exceptionPaths = zone.except || []
           const absoluteFrom = path.resolve(basePath, zone.from)
 
           if (containsPath(absoluteImportPath, absoluteFrom)) {
+            if (exceptionPaths.some((exceptionPath) => {
+              const absoluteExceptionPath = path.resolve(absoluteFrom, exceptionPath)
+              const relativeExceptionPath = path.relative(absoluteFrom, absoluteExceptionPath)
+
+              if (importType(relativeExceptionPath, context) === 'parent') {
```

Comment:
@ljharb this will be the same check every time for a config. Do you know of a rule that does this check only once that I could follow as an example, through caching or some other means? Or is this reasonable?

Reply:
I opted to refactor just to clean up some calls and remove nesting. Otherwise, I figure caching might add unnecessary complexity for what would be a pretty small (albeit repetitive) check.